File: acceptance/sulabhghimire/simplify_path.dart
class Solution {
  String simplifyPath(String path) {
    List<String> stack = [];
    String curr = "";
    path += '/';
    List<String> pathValue = path.split('');

    for (String char in pathValue) {
      if (char == '/') {
        if (curr == '..') {
          if (stack.isNotEmpty) {
            stack.removeLast();
          }
        } else if (curr != '' && curr != '.') {
          stack.add(curr);
        }
        curr = '';
      } else {
        curr += char;
      }
    }

    String val = '/' + stack.join('/');
    return val;
  }
}

File: acceptance/kushal/simplifyPath.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        components = path.split('/')

        for component in components:
            if component == '' or component == '.':
                continue
            elif component == '..':
                if stack:
                    stack.pop()
            else:
                stack.append(component)

        simplified_path = '/' + '/'.join(stack)
        return simplified_path

s = Solution()
print(s.simplifyPath("/home/"))      
print(s.simplifyPath("/../"))          
print(s.simplifyPath("/home//foo/"))   
